server/catgenome/src/test/java/com/epam/catgenome/manager/reference/MotifSearchManagerTestSuccess.java

```diff
@@ -0,0 +1,166 @@
+        referenceTwo = referenceManager.registerGenome(request);
+        idRefTwo = referenceTwo.getId();
+        idChrTwo = referenceTwo.getChromosomes().get(1).getId();
+    }
+
+    @After
+    public void unregisterFiles() throws IOException {
+        referenceManager.unregisterGenome(referenceOne.getId());
+        referenceManager.unregisterGenome(referenceTwo.getId());
+    }
+
+    @Test
+    public void getSomeResultsForSpecifiedPositions() {
+        MotifSearchRequest att = MotifSearchRequest.builder()
+                .referenceId(idRef)
+                .startPosition(1)
+                .endPosition(1000)
+                .chromosomeId(idChr)
+                .motif("AAC")
+                .searchType(MotifSearchType.WHOLE_GENOME)
+                .pageSize(Integer.MAX_VALUE)
+                .strand(StrandSerializable.POSITIVE)
+                .build();
+        MotifSearchResult search = motifSearchManager.search(att);
+        Assert.assertFalse(search.getResult().isEmpty());
+    }
+
+    @Test
+    public void getSomeResultsForSpecifiedPageSize() {
+        MotifSearchRequest att = MotifSearchRequest.builder()
+                .referenceId(idRef)
+                .startPosition(1)
+                .chromosomeId(idChr)
+                .motif("AAC")
+                .searchType(MotifSearchType.WHOLE_GENOME)
+                .pageSize(100)
+                .strand(StrandSerializable.POSITIVE)
+                .build();
+        MotifSearchResult search = motifSearchManager.search(att);
+        Assert.assertEquals(search.getResult().size(), (int) att.getPageSize());
+    }
+
+    @Test
+    public void getResultsFromSeveralChromosomes() {
+        MotifSearchRequest att = MotifSearchRequest.builder()
+                .referenceId(idRefTwo)
+                .chromosomeId(idChrTwo)
+                .motif("AAC")
+                .searchType(MotifSearchType.WHOLE_GENOME)
+                .pageSize(10000)
+                .strand(StrandSerializable.POSITIVE)
+                .build();
+        MotifSearchResult search = motifSearchManager.search(att);
+        Set<String> chromosomes = search.getResult().stream().map(Motif::getContig)
+                .collect(Collectors.toSet());
+        Assert.assertTrue(chromosomes.size() > 1);
+    }
+
+    @Test
+    public void getEmptyPositionForLargeNumberOfResults() {
+        MotifSearchRequest att = MotifSearchRequest.builder()
+                .referenceId(idRefTwo)
+                .startPosition(1)
+                .chromosomeId(idChrTwo)
+                .motif("AAC")
+                .searchType(MotifSearchType.WHOLE_GENOME)
+                .pageSize(10000)
+                .strand(StrandSerializable.POSITIVE)
+                .build();
+        MotifSearchResult search = motifSearchManager.search(att);
+        Assert.assertNull(search.getPosition());
+    }
+
+    @Test
+    public void getResultsFromTwoChrWhereRightDataOnlyInFirstChr() throws IOException {
```

Comment:
name of the test should show what it is testing -> `checkThatPositionIsNullWhenWeSearchFullGenomeButDataOnlyInFirstChrTest`